src/openrct2/world/Wall.cpp

```diff
@@ -35,26 +35,15 @@
  */
 void wall_remove_at(const CoordsXYRangedZ& wallPos)
 {
-    TileElement* tileElement;
-
-repeat:
-    tileElement = map_get_first_element_at(wallPos);
-    if (tileElement == nullptr)
-        return;
-    do
+    auto wallElement = map_get_wall_element_at(wallPos);
```

Comment:
You need to do a while loop on this as we keep searching for wall elements at this coordinate

Reply:
I used a for loop as I think it makes it more readable. Tell me what you think.